test-utils/src/dev/test_cmds.rs

```diff
@@ -125,7 +125,51 @@ pub fn error_for_enoent() -> String {
 /// invocation to invocation (e.g., assigned TCP port numbers, timestamps)
 ///
 /// This allows use to use expectorate to verify the shape of the CLI output.
-pub fn redact_variable(input: &str) -> String {
+#[derive(Clone, Debug)]
+pub struct Redactor<'a> {
+    redact_uuids: bool,
+    extra: ExtraRedactions<'a>,
+}
+
+impl<'a> Redactor<'a> {
+    pub fn new() -> Self {
+        Self { redact_uuids: true, extra: ExtraRedactions::new() }
+    }
+
+    pub fn redact_uuids(&mut self, redact_uuids: bool) -> &mut Self {
+        self.redact_uuids = redact_uuids;
+        self
+    }
+
+    pub fn extra(
+        &mut self,
+        extra_redactions: ExtraRedactions<'a>,
+    ) -> &mut Self {
+        self.extra = extra_redactions;
+        self
+    }
+
+    pub fn do_redact(&self, input: &str) -> String {
+        // Perform extra redactions at the beginning, not the end. This is because
+        // some of the built-in redactions in redact_variable might match a
+        // substring of something that should be handled by extra_redactions (e.g.
```

Comment:
[bikeshed 2/2]:

```suggestion
    pub fn redact(&self, input: &str) -> String {
        // Perform extra redactions at the beginning, not the end. This is because
        // some of the built-in redactions in redact_variable might match a
        // substring of something that should be handled by extra_redactions (e.g.
```

Reply:
I had `redact` at first, but especially now that the config knobs are shorter (e.g. `uuids`, `basic`) I wanted to draw a distinction between the knobs and the actual process of redaction.